src/Libraries/Microsoft.Extensions.AsyncState/AsyncState.cs

```diff
@@ -59,7 +60,7 @@ public bool TryGet(AsyncStateToken token, out object? value)
             return false;
         }
 
-        value = _asyncContextCurrent.Value.Features.Get(token.Index);
+        _ = _asyncContextCurrent.Value.Features.TryGetValue(token, out value);
```

Comment:
I guess we can return the result of `TryGetValue()` call

Reply:
We cannot because false is returned only in if the context is not initialized.